src/components/content/order/CreateService.tsx

```diff
@@ -0,0 +1,369 @@
+    const [flavorOptions, setFlavorOptions] = useState<{ value: string; label: string }[]>([]);
+    const [flavorValue, setFlavorValue] = useState<string>('');
+    //select area
+    const [areaMapper, setAreaMapper] = useState<Map<string, Area[]>>(new Map<string, Area[]>());
+    const [activeKey, setActiveKey] = useState<string>('');
+    const [areaValue, setAreaValue] = useState<string>('');
+    const [areaList, setAreaList] = useState<Area[]>([]);
+    const [items, setItems] = useState<Tab[]>([]);
+    //select region
+    const [regionValue, setRegionValue] = useState<string>('');
+    const [regionOptions, setRegionOptions] = useState<{ value: string; label: string }[]>([{ value: '', label: '' }]);
+
+    const handleChangeVersion = (value: string) => {
+        setVersionValue(value);
+    };
+    const onChangeCloudProvider = (key: string, index: number) => {
+        setCloudProviderValue(key.charAt(0).toLowerCase() + key.slice(1));
+        setIsSelected(index);
+    };
+    const handleChangeFlavor = (value: string) => {
+        setFlavorValue(value);
+    };
+
+    const onChangeAreaValue = (key: string) => {
+        setActiveKey(key);
+        setAreaValue(key);
+    };
+    const handleChangeRegion = (value: string) => {
+        setRegionValue(value);
+    };
+
+    function group(list: any[], key: string): Map<string, any[]> {
+        let map: Map<string, any[]> = new Map<string, any[]>();
+        list.map((val) => {
+            if (!map.has(val[key])) {
+                map.set(
+                    val[key],
+                    list.filter((data) => data[key] === val[key])
+                );
+            }
+        });
+        return map;
+    }
+
+    //set version
+    useEffect(() => {
+        const categoryName = location.search.split('?')[1].split('&')[0].split('=')[1];
+        const serviceName = location.search.split('?')[1].split('&')[1].split('=')[1];
+        if (!categoryName || !serviceName) {
+            return;
+        }
+        setCategoryName(categoryName);
+        setServiceName(serviceName);
+        serviceVendorApi.listRegisteredServices(categoryName, '', serviceName, '').then((rsp) => {
+            if (rsp.length > 0) {
+                let versions: { value: string; label: string }[] = [];
+                const result: Map<string, RegisterServiceEntity[]> = group(rsp, 'version');
+                setVersionMapper(result);
+                result.forEach((v, k) => {
+                    let versionItem = { value: k || '', label: k || '' };
+                    versions.push(versionItem);
+                });
+                setVersionOptions(versions);
+                setVersionValue(versions[0].value);
+                updateArea(versions[0].value, result);
+            } else {
+                return;
+            }
+        });
+    }, [location]);
+
+    //set csp
+    function updateArea(version: string, mapper: Map<string, RegisterServiceEntity[]>): void {
```

Comment:
function name does not reflect the tasks performed by function. 
Also if possible, please split this function.

Reply:
fixed.